clustered/server/src/main/java/org/ehcache/clustered/server/store/ClusterTierActiveEntity.java

```diff
@@ -586,40 +594,74 @@ public void synchronizeKeyToPassive(PassiveSynchronizationChannel<EhcacheEntityM
     if (concurrencyKey == DEFAULT_KEY) {
       stateService.getStateRepositoryManager().syncMessageFor(storeIdentifier).forEach(syncChannel::synchronizeToPassive);
     } else {
+      BlockingQueue<EhcacheDataSyncMessage> messageQ = new SynchronousQueue<>();
       int segmentId = concurrencyKey - DEFAULT_KEY - 1;
-      Long dataSizeThreshold = Long.getLong(SYNC_DATA_SIZE_PROP, DEFAULT_SYNC_DATA_SIZE_THRESHOLD);
-      AtomicLong size = new AtomicLong(0);
-      ServerSideServerStore store = stateService.getStore(storeIdentifier);
-      final AtomicReference<Map<Long, Chain>> mappingsToSend = new AtomicReference<>(new HashMap<>());
-      store.getSegmentKeySets().get(segmentId)
-        .forEach(key -> {
-          final Chain chain;
-          try {
-            chain = store.get(key);
-          } catch (TimeoutException e) {
-            throw new AssertionError("Server side store is not expected to throw timeout exception");
-          }
-          for (Element element : chain) {
-            size.addAndGet(element.getPayload().remaining());
-          }
-          mappingsToSend.get().put(key, chain);
-          if (size.get() > dataSizeThreshold) {
-            syncChannel.synchronizeToPassive(new EhcacheDataSyncMessage(mappingsToSend.get()));
-            mappingsToSend.set(new HashMap<>());
-            size.set(0);
+      CompletableFuture<Void> asyncGet = CompletableFuture.runAsync(() -> doGetsForSync(segmentId, messageQ));
+      try {
```

Comment:
I will let others comment but this will stuck a thread of the common pool for a bit of time. Also, Ehcache configuration is defining thread-pools. Using them would make sense I think.

Reply:
It's not using the common pool though is it?  It's using the `SYNC_GETS_EXECUTOR`.

Reply:
Yes. It's not anymore. The code changed since my comment

Reply:
My bad, sorry.